0278-template-only-components.md

```diff
@@ -0,0 +1,249 @@
+wrapper element in the component template, offering direct control over the
+tag name and other attributes. It would also allow the component to have more
+than one top-level element, or none at all.
+
+In other words, this flag changes template-only components in the app to have
+"Outer HTML" semantics. _What you type is what you get._
+
+Notably, [Glimmer Components](https://glimmerjs.com/guides/templates-and-helpers)
+have adopted the "Outer HTML" semantics long ago and the experience has been
+very positive. This would be one of the first pieces of the Glimmer.js experiment
+to make its way into Ember. We think this feature is small, self-contained but
+useful enough to be integrated back into Ember at this point.
+
+If accepted, this RFC will fully subsume the [Non-context-shifting partials](https://github.com/emberjs/rfcs/pull/262)
+RFC. We can therefore (at a later time, in a separate RFC) explore deprecating
+partials in favor of wrapper-free template-only components.
+
+# Detailed design
+
+## API Surface
+
+We should not expose the flag directly as a public API. Instead, we should
+abstract the flag with a "privileged addon" whose only purpose is to enable
+the flag. Applications will enable the flag by installing this addon. This
+will allow for more flexibility in changing the flag's implementation (the
+location, naming, value, or even the existence of it) in the future. From the
+user's perspective, it is the _addon_ that provides thisfunctionality. The
+flag is simply an internal implementation detail.
+
+We have done this before in other cases (such as the legacy view addon during
+the 2.0 transition period), and it has generally worked well.
+
+When landing this feature, it will be entirely opt-in for existing apps, but
+the Ember CLI application blueprint should be updated to include the addon by
+default. At a later time, we should provide another addon that _disables_ the
+flag explicitly (installing both addons would be an install-time error). At
+that time, we will issue a deprecation warning if the flag is *not set*, with
+a message that directs the user to install one of the two addons.
+
+## Single Global "Flag"
+
+The proposed flag will be truly global in scope. That is, setting this flag
+will change the semantics of all template-only components in the entire app,
+even for components that were included by addons.
+
+However, we believe this would not affect any addon components in practice,
+as the predominant pattern for addons to expose components currently
+necessitates a JavaScript class. Addon authors would create the component
+(with or without a JavaScript class) in the `/addon` folder, but exposing
+it for consumption in apps requires creating a corresponding JavaScript class
+in the `/app` folder to "re-export" the component. Therefore, in practice,
+it is not actually possible for addons to have a truely template-only
+component today (something to address in a future RFC).
```

Comment:
it might still make sense to allow white/blacklisting to ease the transition from with-wrapper to no-wrapper in existing large-scale apps 🤔

Reply:
I just saw that a workaround for this is proposed in the "Migration" section below, which seems fine to me 👍